Errors when running multiple instances of data-prepper in the sample app

Encountering ES client errors:

Example:
```
Caused by: java.lang.RuntimeException: method [PUT], host [https://node-0.example.com:9200], URI [/_opendistro/_ism/policies/raw-span-policy], status line [HTTP/1.1 409 Conflict]
{"error":{"root_cause":[{"type":"version_conflict_engine_exception","reason":"[raw-span-policy]: version conflict, document already exists (current version [1])","index_uuid":"IdnxdC94Qaa7CYJwSPESxg","shard":"0","index":".opendistro-ism-config"}],"type":"version_conflict_engine_exception","reason":"[raw-span-policy]: version conflict, document already exists (current version [1])","index_uuid":"IdnxdC94Qaa7CYJwSPESxg","shard":"0","index":".opendistro-ism-config"},"status":409}
        at com.amazon.dataprepper.plugins.sink.elasticsearch.ElasticsearchSink.<init>(ElasticsearchSink.java:62)
        ... 19 more
Caused by: org.elasticsearch.client.ResponseException: method [PUT], host [https://node-0.example.com:9200], URI [/_opendistro/_ism/policies/raw-span-policy], status line [HTTP/1.1 409 Conflict]
{"error":{"root_cause":[{"type":"version_conflict_engine_exception","reason":"[raw-span-policy]: version conflict, document already exists (current version [1])","index_uuid":"IdnxdC94Qaa7CYJwSPESxg","shard":"0","index":".opendistro-ism-config"}],"type":"version_conflict_engine_exception","reason":"[raw-span-policy]: version conflict, document already exists (current version [1])","index_uuid":"IdnxdC94Qaa7CYJwSPESxg","shard":"0","index":".opendistro-ism-config"},"status":409}
....
2523 [main] ERROR com.amazon.dataprepper.parser.PipelineParser  – Construction of pipeline components failed, skipping building of pipeline [raw-pipeline]

```

So far I've hit issues in ElasticsearchSink.java and IndexStateManagement.java, both around the lowLevelClient performing requests. Need to handle these errors gracefully instead of failing the pipeline initialization completely.

--- a/data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java
+++ b/data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java
@@ -5,6 +5,7 @@ import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
 import com.amazon.dataprepper.model.configuration.PluginSetting;
 import com.amazon.dataprepper.model.record.Record;
 import com.amazon.dataprepper.model.sink.Sink;
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.action.DocWriteRequest;
 import org.elasticsearch.action.admin.indices.alias.Alias;
 import org.elasticsearch.action.admin.indices.alias.get.GetAliasesRequest;
@@ -15,7 +16,6 @@ import org.elasticsearch.client.RestHighLevelClient;
 import org.elasticsearch.client.indices.CreateIndexRequest;
 import org.elasticsearch.client.indices.GetIndexRequest;
 import org.elasticsearch.client.indices.PutIndexTemplateRequest;
-
 import org.elasticsearch.common.unit.ByteSizeUnit;
 import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
@@ -183,7 +183,16 @@ public class ElasticsearchSink implements Sink<Record<String>> {
         initialIndexName = indexAlias;
         createIndexRequest = new CreateIndexRequest(initialIndexName);
       }
-      restHighLevelClient.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+      try {
+        restHighLevelClient.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+      } catch (ElasticsearchException e) {
+        if (e.getMessage().contains("resource_already_exists_exception")) {
+          // Do nothing - likely caused by a race condition where the resource was created
+          // by another host before this host's restClient made its request
+        } else {
+          throw e;
+        }
+      }
     }
   }
 

--- a/data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/IndexStateManagement.java
+++ b/data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/IndexStateManagement.java
@@ -5,6 +5,7 @@ import org.elasticsearch.action.admin.cluster.settings.ClusterGetSettingsRespons
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
 import org.elasticsearch.client.RestHighLevelClient;
 import org.elasticsearch.rest.RestStatus;
 
@@ -41,7 +42,17 @@ public class IndexStateManagement {
                 is.close();
                 request = new Request(HttpMethod.PUT, endPoint);
                 request.setJsonEntity(policyJsonBuffer.toString());
-                restHighLevelClient.getLowLevelClient().performRequest(request);
+                try {
+                    restHighLevelClient.getLowLevelClient().performRequest(request);
+                } catch (ResponseException e) {
+                    if (e.getMessage().contains("version_conflict_engine_exception")
+                            || e.getMessage().contains("resource_already_exists_exception")) {
+                        // Do nothing - likely caused by a race condition where the resource was created
+                        // by another host before this host's restClient made its request
+                    } else {
+                        throw e;
+                    }
+                }
             }
             return IndexConstants.RAW_ISM_POLICY;
         } else {

